lib/rules/prefer-read-only-props.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview Require component props to be typed as read-only.
+ * @author Luke Zapart
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const docsUrl = require('../util/docsUrl');
+
+function isFlowPropertyType(node) {
+  return node.type === 'ObjectTypeProperty';
```

Comment:
does this conflict with TS node names at all?

Reply:
Good question. I checked and AST that `typescript-eslint-parser` generates for `function Hello(props: {name: string}) {...}` is: `FunctionDeclaration <- TSTypeAnnotation <- TSTypeLiteral <- TSPropertySignature <- TSStringKeyword`, so it's safe.